docker/test_env/genesis_generator.go

```diff
@@ -0,0 +1,157 @@
+package test_env
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/docker"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+)
+
+type EthGenesisGeneretor struct {
+	EnvComponent
+	chainConfig          EthereumChainConfig
+	l                    zerolog.Logger
+	generatedDataHostDir string
+	t                    *testing.T
+}
+
+func NewEthGenesisGenerator(chainConfig EthereumChainConfig, generatedDataHostDir string, opts ...EnvComponentOption) *EthGenesisGeneretor {
+	g := &EthGenesisGeneretor{
+		EnvComponent: EnvComponent{
+			ContainerName: fmt.Sprintf("%s-%s", "eth-genesis-generator", uuid.NewString()[0:8]),
+		},
+		chainConfig:          chainConfig,
+		generatedDataHostDir: generatedDataHostDir,
+		l:                    log.Logger,
+	}
+	for _, opt := range opts {
+		opt(&g.EnvComponent)
+	}
+	return g
+}
+
+func (g *EthGenesisGeneretor) WithTestInstance(t *testing.T) *EthGenesisGeneretor {
+	g.l = logging.GetTestLogger(t)
+	g.t = t
+	return g
+}
+
+func (g *EthGenesisGeneretor) StartContainer() error {
+	r, err := g.getContainerRequest(g.Networks)
+	if err != nil {
+		return err
+	}
+
+	l := logging.GetTestContainersGoTestLogger(g.t)
+	_, err = docker.StartContainerWithRetry(g.l, tc.GenericContainerRequest{
+		ContainerRequest: *r,
+		Reuse:            true,
+		Started:          true,
+		Logger:           l,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "cannot start eth genesis generation container")
+	}
+
+	g.l.Info().Str("containerName", g.ContainerName).
+		Msg("Started Eth Genesis container")
+
+	return nil
+}
+
+func (g *EthGenesisGeneretor) getContainerRequest(networks []string) (*tc.ContainerRequest, error) {
+	valuesEnv, err := os.CreateTemp("", "values.env")
+	if err != nil {
+		return nil, err
+	}
+
+	bc, err := generateEnvValues(&g.chainConfig)
+	if err != nil {
+		return nil, err
+	}
+	_, err = valuesEnv.WriteString(bc)
+	if err != nil {
+		return nil, err
+	}
+
+	elGenesisFile, err := os.CreateTemp("", "genesis-config.yaml")
+	if err != nil {
+		return nil, err
+	}
+	_, err = elGenesisFile.WriteString(elGenesisConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	clGenesisFile, err := os.CreateTemp("", "config.yaml")
+	if err != nil {
+		return nil, err
+	}
+	_, err = clGenesisFile.WriteString(clGenesisConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	mnemonicsFile, err := os.CreateTemp("", "mnemonics.yaml")
+	if err != nil {
+		return nil, err
+	}
+	_, err = mnemonicsFile.WriteString(mnemonics)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tc.ContainerRequest{
+		Name:          g.ContainerName,
+		Image:         "tofelb/ethereum-genesis-generator:2.0.4-slots-per-epoch",
```

Comment:
Need to ask about moving this to our ECR

Reply:
I've opened a PR to the original repo, maybe they will merge it. If not, I'd create a private repo in our org, copy my fork there, then build a private image based on that and push it to our ECR. WDYT?